### ISSUE TITLE
Problem: Python binding is broken

### ISSUE THREAD

**Issue by hintjens** [zeromq/zproject#408]
The troublemaker is zframe. It's unclear how to handle the argument. It used to be "anything", but some bindings (e.g. JNI) need specific handling for byte arrays. So I changed this to "buffer". That breaks the Python binding:

```
Traceback (most recent call last):
  File "bindings/python/test.py", line 261, in test_zframe
    frame = Zframe(b"Hello", 5)
  File "/home/ph/work/hintjens/czmq/bindings/python/czmq.py", line 1402, in __init__
    self._as_parameter_ = lib.zframe_new(args[0], args[1]) # Creation of new raw type
ArgumentError: argument 1: <type 'exceptions.TypeError'>: expected LP_c_byte instance instead of str
```

**Comment by twhittock** [comment]
Probably fixable by doing something like: `frame = Zframe(create_string_buffer("Hello"), 5)`

see also other create_string_buffer calls. It's not as nice, but the bindings aren't really meant to be idiomatic so whatever.

**Comment by hintjens** [comment]
I tried several times to get czmq's test.py to work, and failed. It's all voodoo to me. If you have a moment to fix this, that'd be great. The alternative is to strip out all hand-written Python tests and simply call the test method on each class.

**Comment by twhittock** [comment]
Finally fixed in zeromq/czmq#1485

**Comment by bluca** [comment]
Closing since it was fixed - thanks!
